packages/google_maps_flutter/lib/src/marker_updates.dart

```diff
@@ -0,0 +1,186 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of google_maps_flutter;
+
+/// Types of [Marker] updates.
+enum MarkerUpdateEventType {
+  add,
+  update,
+  remove,
+}
+
+Map<MarkerId, Marker> _toMap(Set<Marker> markers) {
+  final Map<MarkerId, Marker> result = <MarkerId, Marker>{};
+  if (markers == null) {
+    return null;
+  }
+  markers.forEach((Marker marker) {
+    if (marker != null) {
+      result[marker.markerId] = marker;
+    }
+  });
+  return result;
+}
+
+/// [Marker] update events to be applied to the [GoogleMap].
+///
+/// Used in [GoogleMapController] when the map is updated.
+@immutable
+class MarkerUpdates {
```

Comment:
Is this essentially a set of MarkerUpdate objects? why do we need a dedicated type for it?

Reply:
This does a few more things than just being a wrapper:

1. This class provides a nice place to have `.from(previous, current)` to compute the diff.
2. Provides methods for serialization of the update chunk that we need to send over to the platform side.
3. Though this is a wrapper around a `Set<MarkerUpdate>`, it succinctly captures a snapshot of all updates done to the markers.